kratos/tests/containers/test_variables.cpp

```diff
@@ -0,0 +1,51 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license:
+//kratos/license.txt
+//
+//  Main authors:    Pooyan Dadvand
+//
+//
+
+// System includes
+#include <unordered_set>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "includes/variables.h"
+
+namespace Kratos {
+namespace Testing {
+
+// This test is to check the hash function used in variable key, if it fails 
+// means that we should change the hash function. Pooyan.
+KRATOS_TEST_CASE_IN_SUITE(VariablesKeyUniqueness, KratosCoreFastSuite) {
+  std::unordered_set<VariableData::KeyType> registered_keys;
+  for (auto &i : KratosComponents<VariableData>::GetComponents()) {
+    auto hash = i.second->Key();
+    if (registered_keys.find(hash) == registered_keys.end()) {
+      registered_keys.insert(hash);
+    } else {
+      KRATOS_ERROR << "Duplicated variables key founded for variable" << *(i.second) << std::endl;
+    }
+  }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(VariablesKeyOrder, KratosCoreFastSuite) {
+    KRATOS_CHECK_EQUAL(VELOCITY_X.Key() + 1, VELOCITY_Y.Key()) << VELOCITY_X << " , " << VELOCITY_Y;
```

Comment:
I would suggest to loop over all the variables, and if it is component verify if this is verified. Otherwise it may be ok for VELOCITY but not so for say MESH_VELOCITY

Reply:
I see the problem you are mentioning but still I don't know which variables have which components to check them for order. Do you have a solution in mind?

Reply:
a component variable knows the name of the string variable, as well as its component index. You also have is_component.
Using this 3 things it should be possible to write the test in a generic way

Reply:
I see, let me check what can I do.